staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1255,23 +1255,28 @@ func (c *cacheWatcher) nextBookmarkTime(now time.Time, bookmarkFrequency time.Du
 	return heartbeatTime, true
 }
 
-func getEventObject(object runtime.Object) runtime.Object {
+func getMutableObject(object runtime.Object) runtime.Object {
 	if _, ok := object.(runtime.CacheableObject); ok {
```

Comment:
pre-existing, but I'm wondering why we're checking an interface here and a concrete type (`*cachingObject`) below... the concrete type check seems safer

Reply:
You mean in updateResourceVersionIfNeeded?

I think we should switch to CacheableObject there as it's more generic, no?

Reply:
implementing the CacheableObject interface doesn't actually guarantee it is safe to mutate... we know the concrete type is safe to mutate

Reply:
ok - done